### ISSUE TITLE
Send e-mail to the reporter after rejecting one or multiple versions

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-server#21760]
STR:

1. Reported an add-on and report went into the dev addon infringement queue 
2. From rev tools the flagged version is rejected + check mark on the DSA report 

Result:
checking e-mails, one is sent to the reported about the version/add-on being disabled but there's not e-mail sent to reporter about 
content being removed 

Expected result:
an e-mail should be sent to reporter too when rejecting one or multiple versions

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDSRV-677)

**Comment by eviljeff** [comment]
Ah, the reporter email for rejecting versions is only added with https://github.com/mozilla/addons-server/pull/21755 so won't work until that lands

**Comment by ioanarusiczki** [comment]
@eviljeff 

E-mail is sent to the reporter now when a version is rejected. 

I've some question marks:

- when I use delay rejection the reporter email saying the content was taken down is sent right after the action. When rejection is delayed it goes to Pending rejections where the options would be Reject, Force disable, block or Clear pending rejection. I think the reporter emails should be sent only after a decision is taken in Add-ons Pending Rejection queue. 

- for an extension with multiple versions and all approved, it is not installed by the reporter, and it gets rejected because of being flagged for human review with "Reported for abuse within the add-on". The add-on is reported again (no version is installed), the flag for human review is added to the [rejected version](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620115) again and not the current approved/available version.

- Reporter is sending a report for an add-on with mixed versions from the detail page of the listed version. The listed and the unlisted are[ both flagged](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/629446)

**Comment by eviljeff** [comment]
> I've some question marks:
> 
>     * when I use delay rejection the reporter email saying the content was taken down is sent right after the action. When rejection is delayed it goes to Pending rejections where the options would be Reject, Force disable, block or Clear pending rejection. I think the reporter emails should be sent only after a decision is taken in Add-ons Pending Rejection queue.

Mmm, I see your point, though we don't have the necessary emails to handle the scenario where it's delayed rejected (so the reporter's report was valid), but then the version _isn't_ rejected because a new version is submitted or they made the necessary changes.  cc @wagnerand this is a follow-up scenario I guess we will want to address?

>     * for an extension with multiple versions and all approved, it is not installed by the reporter, and it gets rejected because of being flagged for human review with "Reported for abuse within the add-on". The add-on is reported again (no version is installed), the flag for human review is added to the [rejected version](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620115) again and not the current approved/available version.

Currently we just flag the (absolute) latest version if we don't get an installed version.  @wagnerand to advise if this warrants a follow-up issue.

>     * Reporter is sending a report for an add-on with mixed versions from the detail page of the listed version. The listed and the unlisted are[ both flagged](https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/629446)

That's by design - if we don't have the installed version we don't know which channel they're reporting, so both are flagged.

**Comment by wagnerand** [comment]
> Mmm, I see your point, though we don't have the necessary emails to handle the scenario where it's delayed rejected (so the reporter's report was valid), but then the version isn't rejected because a new version is submitted or they made the necessary changes. cc @wagnerand this is a follow-up scenario I guess we will want to address?

Yes, could you work with Ashley to get us a separate email copy for a delayed rejection?

> Currently we just flag the (absolute) latest version if we don't get an installed version. @wagnerand to advise if this warrants a follow-up issue.

Hm, if we don't have version information, I think we should flag the latest public version, for listed at least. For unlisted, most if not all reports should come from Firefox which passes a version. But if we don't have that, I'm not sure what to flag. We could flag the absolute latest version, but we need to somehow indicate to the reviewer that all versions should be looked at.

**Comment by eviljeff** [comment]
> > Currently we just flag the (absolute) latest version if we don't get an installed version. @wagnerand to advise if this warrants a follow-up issue.
> 
> Hm, if we don't have version information, I think we should flag the latest public version, for listed at least. For unlisted, most if not all reports should come from Firefox which passes a version. But if we don't have that, I'm not sure what to flag. We could flag the absolute latest version, but we need to somehow indicate to the reviewer that all versions should be looked at.

In the follow-up issue someone is going to need to specify what "somehow indicate to the reviewer that all versions should be looked at" looks like.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-677
